src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,143 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MIN_SIZE = 10;
+    private int arraySize;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        arraySize = MIN_SIZE;
+        elementData = new Object[arraySize];
+        size = 0;
+    }
+
+    private void grow() {
+        Object[] oldArray = elementData;
+        elementData = new Object[(int) (oldArray.length + (oldArray.length * 0.5))];
```

Comment:
create a GROW_FACTOR of 1.5

Reply:
I did not fully understand the comment, please explain